massiv/src/Data/Massiv/Array/Manifest/BoxedNF.hs

```diff
@@ -35,23 +39,36 @@ import           Data.Massiv.Core.Common
 import           Data.Massiv.Core.List
 import           Data.Massiv.Core.Scheduler
 import qualified Data.Primitive.Array                as A
+import           Data.Validity
 import qualified Data.Vector                         as VB
 import qualified Data.Vector.Mutable                 as VB
 import           GHC.Exts                            as GHC (IsList (..))
 import           Prelude                             hiding (mapM)
 import           System.IO.Unsafe                    (unsafePerformIO)
+import           GHC.Generics (Generic)
 
 -- | Array representation for Boxed elements. This structure is element and
 -- spine strict, and elements are always in Normal Form (NF), therefore `NFData`
 -- instance is required.
-data N = N deriving Show
+data N = N deriving (Show, Generic)
+
+instance Validity N
 
 type instance EltRepr N ix = M
 
+instance Validity a => Validity (A.Array a) where
+#if MIN_VERSION_primitive(0,6,2)
+    validate = foldMap validate
+#else
+    validate = trivialValidation
```

Comment:
Again: this is wrong, but no need to fix it if you can replace the dependency with the newest version of `validity-array` soon.